### PR TITLE
Initial NodeIsPromoted condition implementation.

### PR DESCRIPTION
**Pull request by paranojik**
Implementation of https://drupal.org/node/2269565 following other condition implementations.

### CODE REVIEW

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * Defines a rule condition for promoted node.
+ *
+ * @Condition(
+ *   id = "rules_node_is_promoted",
+ *   label = @Translation("Node is promoted"),
+ *   context = {
+ *     "node" = {
+ *       "type" = "entity:node"
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: This also needs a label.

File: src/Tests/Condition/NodeIsPromotedTest.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Tests\Plugin\Condition.
```
Comment (fubhy) [inline src/Tests/Condition/NodeIsPromotedTest.php]: Needs a leading backslash.

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Plugin\Condition\NodeIsPromoted.
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: Needs a leading backslash (\Drupal...)

**Comment by fubhy** [comment]
Good work. I left a couple of in-line notes. Once those are fixes this is ready to be merged.

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+
+/**
+ * Defines a rule condition for promoted node.
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: Let's make this uniform with the other conditions:

"Provides a 'Node is promoted' condition."

**Comment by fubhy** [comment]
Oh. I just saw that there's also this here: https://github.com/fago/rules/pull/6 which basically this is a duplicate of. Also, you need to update this pull requests as the core context classes and interfaces are now slightly different as fago merged this: https://github.com/fago/rules/commit/c8aa15bee242084ba80a91db5d2d9512e3821a43 yesterday.

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add group information from Drupal 7.
+ */
+class NodeIsPromoted extends ConditionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return t('Node is promoted');
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: Should be $this->t('Node is promoted');

**Comment by paranojik** [comment]
Yup. I asked about this in the issue queue on dgo. So... which is the way
to go?

2014-05-30 14:55 GMT+02:00 fubhy notifications@github.com:

File: src/Tests/Condition/NodeIsPromotedTest.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->conditionManager = $this->container->get('plugin.manager.condition', $this->container->get('container.namespaces'));
+    $this->nodeStorage = $this->entityManager->getStorage('node');
+
+    $this->entityManager->getStorage('node_type')
+      ->create(array('type' => 'page'))
```
Comment (fubhy) [inline src/Tests/Condition/NodeIsPromotedTest.php]: Sorry.. One more thing. Can you please switch from using array() to [] instead? We just recently merged a PR that converted all occurrences as we are going to try and stick to the short array syntax ([]).

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+ * Contains \Drupal\rules\Plugin\Condition\NodeIsPromoted.
+ */
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\rules\Context\ContextDefinition;
+use Drupal\rules\Engine\RulesConditionBase;
+
+/**
+ * Provides a 'Node is promoted' condition..
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: Two periods in that line :)

File: src/Plugin/Condition/NodeIsPromoted.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * @Condition(
+ *   id = "rules_node_is_promoted",
+ *   label = @Translation("Node is promoted"),
```
Comment (fubhy) [inline src/Plugin/Condition/NodeIsPromoted.php]: That last comma should not be there.

**Comment by fubhy** [comment]
Sorry, I left you 3 more notes. Ready to be merged after those. Thanks!

**Comment by fubhy** [comment]
Perfect, thanks! Now, if you could just remove the comment "Another round of @fubhy..." from the commit message that would be awesome - Then we can merge this :).

To do so, just do "git commit --amend" remove the commit comment, then git push --force :).

Thanks!

**Comment by paranojik** [comment]
:) Thanks for you patience @fubhy.

**Comment by fubhy** [comment]
Perfect. This is ready to go in. Now we just have to wait for fago or klausi to merge it :).

**Pull request merged**